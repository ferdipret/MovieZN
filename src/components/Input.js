import React from 'react'
import styled from 'styled-components'

const Input = ({ className, onSearchInputChange, onSearchSubmit, value }) => (
  <form className={className} onSubmit={onSearchSubmit}>
    <input
      className="search-input"
      onChange={onSearchInputChange}
      value={value}
    />
    <button type="submit" className="form-submit-btn hidden" />
  </form>
)

export default styled(Input)`
  .hidden {
    display: none;
  }
`
