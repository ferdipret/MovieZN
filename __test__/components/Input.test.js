import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, find, simulate } from 'enzyme'

import Input from 'components/Input'

describe('Input', () => {
  test('should render correctly', () => {
    const rendered = renderer.create(<Input />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
