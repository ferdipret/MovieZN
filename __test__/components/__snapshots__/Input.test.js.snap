// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<form
  className="sc-bdVaJa cLDRn"
  onSubmit={undefined}
>
  <input
    className="search-input"
    onChange={undefined}
    value={undefined}
  />
  <button
    className="form-submit-btn hidden"
    type="submit"
  />
</form>
`;
